### PR TITLE
[Publisher][Superagencies] Add dropdown for child agencies to Explore Data page

### LINKED ISSUE #1276: [Superagencies][3/n] Add toggle for child agencies in Explore Data
Now that we're saving the right data in Agency Store, no matter where we are in the app, if the user is logged into to a superagency, we should know which child agencies belong to that superagency.

We'd like to add a dropdown on the Explore Metrics page that looks like [this Figma](https://www.figma.com/file/dvX3yg7FMGJSWFaqHWrWgQ/Justice-Counts-Publisher?type=design&node-id=10942-8061&mode=design&t=LDKleuhKEP9M2NJu-0). Note that you should look at the second Figma, the one that says "Old Explore Data UI", since we have not made the update yet.

The dropdown should only appear if the current agency is a superagency, and it should be populated with that agency's child agencies in alphabetical order. It should also be searchable.

When the user selects a child agencies, the page should behave similar to what would happen just like if the user had used the menu in the top left to switch to that child agency (but only this page is refreshed).

**If the child agencies differ in the systems they belong to** -- for instance if some of the child agencies belong to Law Enforcement and some to Courts -- we want to add _sections_ to the dropdown and group the child agencies by which systems they belong to. They should be sorted alphabetically within each section. Reach out to @mxosman  and @forrestkim00 when you get to this part for more clarity!

### PR DESCRIPTION
**Pull request by nasaownsky**
## Description of the change

1. Added separate ChildAgenciesDropdown component 
2. Added ChildAgenciesDropdown to explore data page

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

closes #1276 

### Development

**This box MUST be checked by the submitter prior to merging**:
- [ ] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by mxosman** [comment]
> @mxosman Maybe you know what is this lint `Error: 18:1 error incorrect header header/header` in CI tests?

Ah - we probably haven't updated the year in our license header in all of our files, and the new file you added shows 2024. For now, can you see if `yarn lint --fix` will take care of the adjustment?

File: publisher/src/components/MetricsConfiguration/ChildAgenciesDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+          size="small"
+          caretPosition="right"
+          fullWidth
+          typeaheadSearch={{ placeholder: "Search for Agency" }}
+          customClearSearchButton="Clear"
+        />
+      </Styled.DropdownWrapper>
+    )
+  );
+});
```
Comment (mxosman) [inline publisher/src/components/MetricsConfiguration/ChildAgenciesDropdown.tsx:90]: I love everything about this - fantastic call turning this into its own component!

File: publisher/src/components/MetricsConfiguration/MetricsOverview.tsx [hunk 2]
```diff
@@ -109,38 +103,6 @@ export const MetricsOverview = observer(() => {
...
-      ? superagencyChildAgencies
-          .map((agency) => ({
-            key: agency.id,
-            label: agency.name,
-            onClick: () => navigate(`/agency/${agency.id}/metric-config`),
-            highlight: agency.id === currentAgency?.id,
-          }))
-          .sort((a, b) => a.label.localeCompare(b.label))
-      : [];
-
```
Comment (mxosman) [inline publisher/src/components/MetricsConfiguration/MetricsOverview.tsx:143]: 🧹 🧹 🧹

**Comment by mxosman** [comment]
Love the consolidation here, @nasaownsky! It looks & works great for me! One small visual thing I noticed for agencies with a few available metrics - is there a good way to have the left-side (Law Enforcements > Arrest) not be centered - but be in a flex-start-like position (aligned towards the top rather than center)?

Screenshot 2024-04-23 at 9 19 48 AM

File: publisher/src/components/MetricsConfiguration/ChildAgenciesDropdown.tsx (hunk 1 above)
Comment (nasaownsky) [reply publisher/src/components/MetricsConfiguration/ChildAgenciesDropdown.tsx:90]: Thank you so much :)

**Comment by nasaownsky** [comment]
@mxosman certainly, fixes applied!

**Review by mxosman — APPROVED** [review]
Thank you so much, @nasaownsky!

**Comment by helperbot-recidiviz** [comment]
@brandon-hills Playtest deployment failed. See workflow failure logs at https://github.com/Recidiviz/justice-counts/actions/runs/8819915695

**Comment by helperbot-recidiviz** [comment]
@brandon-hills successfully triggered a playtest deployment. Full deployment usually takes 5 minutes. Your playtest link is https://brandontest---publisher-web-b47yvyxs3q-uc.a.run.app/

**Pull request merged**